src/bin/pg_autoctl/fsm_transition.c

```diff
@@ -753,6 +753,103 @@ fsm_stop_postgres(Keeper *keeper)
 }
 
 
+/*
+ * fsm_stop_postgres_for_maintenance is used when pg_autoctl enable maintenance
+ * has been used on the primary server, we do a couple CHECKPOINT before
+ * stopping Postgres to ensure a smooth transition.
+ */
+bool
+fsm_stop_postgres_for_maintenance(Keeper *keeper)
```

Comment:
Should we maybe also call this when going to maintenance as secondary? Otherwise it might be goodbetter to rename the function to make clear that it's only meant for primary maintenance.

Reply:
I don't believe it's necessary to CHECKPOINT a secondary before going to maintenance. Also, we don't even stop Postgres there.